### DISCUSSION TITLE
RTL-SDR development files not found

### DISCUSSION THREAD

**Discussion by VK1IAN** [merbanan/rtl_433#3095] [General]
I'm trying to build rtl_433 on a (old) Pi model B running Raspian 7 (wheezy). rtl-sdr is not available on the repos, so I'm compiling both from source. I managed to compile rtl-sdr (instructions here: https://osmocom.org/projects/rtl-sdr/wiki/Rtl-sdr) and got rtl_test running as expected, however when I trying to compile rtl_433, I get the following message:
"CMake Error at CMakeLists.txt:223 (message):
  RTL-SDR development files not found."
Assuming the DEV files for rtl-sdr have been installed when compiling (not sure how I check this), is there something I need to pass to rtl_433??

**Comment by zuckschwerdt** [comment]
You need to install rtl-sdr pkgconfig files or have the includes in /usr/include or /usr/local/include.
A standard install should give you files like:
/usr/local/bin/rtl_test
/usr/local/include/rtl-sdr.h
/usr/local/lib/librtlsdr.a
/usr/local/lib/librtlsdr.so
/usr/local/lib/pkgconfig/librtlsdr.pc

**Reply by VK1IAN** [reply]
Thanks for the prompt reply. My findings are as follows.
I see nothing that relates (I think) in /usr/include, but I do have bcm2835.h, rtl-sdr_export.h and rtl-sdr.h in /usr/local/include. They are all owned by root but readable to everyone.
I also have all the other files in the locations you specified, although librtlsdr.so is linked to librtlsdr.so.0 which is linked to librtlsdr.so.0.5git; all in the same directory (/usr/local/lib/). Now my head hurts 🥴
Is there something else I need to link or find??

**Reply by zuckschwerdt** [reply]
That seems complete. Not sure why it wouldn't work, maybe because of old pkgconfig or cmake versions? It seems that some archive sites still have Wheezy installers, maybe we can build a wheezy test image.

**Reply by VK1IAN** [reply]
Not sure what that means but I'm happy to help.
This old model B has been in my drawer for ages and I'm loathed to throw it out (nostalgic reasons), so keen to give it a new lease on life.
In the meantime, I might have a scratch around to see if there are any later raspian versions that will run on this old hardware.

**Reply by zuckschwerdt** [reply]
It's a problem with the too old CMake version (missing _FOUND in pkg_check_modules). And then too old libusb (missing libusb_strerror).
But this should work around that, instead of `cmake ..` use `cmake -DLibRTLSDR_FOUND=1 -DLibUSB_FOUND=0 ..`

This is on a fresh Wheezy install with libusb-1.0.0 and pkg-config added, i.e. after running:
```
sudo apt-get install gcc cmake git make autoconf libtool libusb-1.0-0-dev pkg-config
git clone https://github.com/osmocom/rtl-sdr.git
cd rtl-sdr
libtoolize && aclocal && autoheader && automake --add-missing && autoconf
./configure
make
sudo make install
cd
git clone https://github.com/merbanan/rtl_433.git
cd rtl_433
mkdir build ; cd build ; cmake -DLibRTLSDR_FOUND=1 -DLibUSB_FOUND=0 ..
make
make install
```

If you get `rtl_433: error while loading shared libraries: librtlsdr.so.0: cannot open shared object file: No such file or directory` then your LD might not be configured, run `sudo ldconfig`

**Reply by VK1IAN** [reply]
Thanks mate. Love your work. Process took a while on my dinosaur but I'm up and running and decoding data from my weather stations.
A few observations:
- Only thing that got installed with the first command was autoconf. My previous attempts must have installed the rest.
- rtl-sdr compiled without errors or warnings.
- rtl_433 threw a lot of warnings during the make (mainly "missing initializer"), but did complete without errors. I have the whole process output in a text file if you are interested.
- I changed the last command to "sudo make install" since that is what I've been doing in the past.

Again, many thanks for your efforts and advice. I'm a very happy camper 😁
I'll close this discussion as solved 👍

**Reply by zuckschwerdt** [reply]
Glad to hear it works. Yes, `sudo make install` is correct. The compiler warnings are fixed with newer version of GCC, no need to worry ;)
